plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal/Completion/Completer/Paths.hs

```diff
@@ -45,6 +46,26 @@ data PathCompletionInfo = PathCompletionInfo
   }
   deriving (Eq, Show)
 
+
+{- | Posix.splitFileName modification, that drops trailing ./ if
+  if wasn't present in the original path.
+
+  Fix for the issue #3774
+   
+  Examples of path splitting:
+  ""            -> ("", "") instead of ("./","")
+  "./"          -> ("./", "")
+  "dir"         -> ("", "dir") instead of ("./","dir")
+  "./dir"       -> ("./", "dir")
+  "dir1/dir2"   -> ("dir1/","dir2")
+  "./dir1/dir2" -> ("./dir1/","dir2")
+-}
+splitFileNameNoTrailingSlash :: FilePath -> (String, String)
+splitFileNameNoTrailingSlash prefix = rmTrailingSlash ("./" `List.isPrefixOf` prefix) (Posix.splitFileName prefix)
+  where rmTrailingSlash hadTrailingSlash (queryDirectory', pathSegment')
+                    | hadTrailingSlash = (queryDirectory', pathSegment')
+                    | otherwise        = ("./" `dropPrefix` queryDirectory', pathSegment')
```

Comment:
Nitpick: you can achieve the same without depending on stuff from extra (not that it's bad)

```suggestion
splitFileNameNoTrailingSlash :: FilePath -> (String, String)
splitFileNameNoTrailingSlash prefix = dropDotSlash $ Posix.splitFileName prefix
  where
    dropDotSlash ('.':'/':queryDirectory, pathSegment) = (queryDirectory, pathSegment)
    dropDotSlash other = other
```

Reply:
The suggestion is slightly incomplete, as it doesn't keep the `./` if the user explicitly wrote it. Otherwise, we could likely just use `makeRelative "./"` 

```
> makeRelative "./" "./src/Lib"
"src/Lib"
```

I am fine with either implementations :)

Reply:
Ah, I missed this requirement. Ignore my suggestion then.